src/Silex/Provider/ValidatorServiceProvider.php

```diff
@@ -46,7 +48,21 @@ public function register(Container $app)
         };
 
         $app['validator.mapping.class_metadata_factory'] = function ($app) {
-            return new LazyLoadingMetadataFactory(new StaticMethodLoader());
+            $loader = null;
+
+            if ($app->offsetExists('validator.mapping.use_annotation') && $app['validator.mapping.use_annotation'] === true) {
+                $loader = new AnnotationLoader(new AnnotationReader());
+            } else {
+                $loader = new StaticMethodLoader();
+            }
+
+            $cache = null;
+
+            if ($app->offsetExists('validator.mapping.cache')) {
```

Comment:
1. I think code preference doesn't add anything. But if you read carefully, you will notice that the method `offsetExists` doesn't try to retrieve the value, just check if its there.

2. Well, yes, A default value won't hurt, but in this case I want it to be null anyway, so it doesn't make much sense to me.